rusk-profile/src/lib.rs

```diff
@@ -108,14 +108,9 @@ pub fn get_rusk_state_dir() -> Result<PathBuf, io::Error> {
         })
 }
 
-pub fn get_rusk_state_id_path() -> Result<PathBuf, io::Error> {
-    let mut state_dir = get_rusk_state_dir()?;
-    state_dir.set_file_name("state.id");
-    Ok(state_dir)
-}
-
-pub fn get_rusk_state_id() -> Result<Vec<u8>, io::Error> {
-    read(get_rusk_state_id_path()?)
+pub fn to_rusk_state_id_path<P: AsRef<Path>>(dir: P) -> PathBuf {
+    let dir = dir.as_ref();
+    dir.join("state.id")
```

Comment:
Yes, this is on purpose. The "current ID" is a part of the state, and as such should be in the state directory. This no longer works like `microkelvin` where we could point to whatever root is appropriate. Keep in mind that, as mentioned during a long meeting, this "state.id" will dissappear in time anyway.

As for the method structure, I tend to agree that you approach is nicer, but in this case I did it to avoid cloning paths all the time.